src/client.js

```diff
@@ -148,15 +145,31 @@ assign(Client.prototype, {
 
   Pool: Pool2,
 
-  initializePool: function(config) {
-    if (this.pool) this.destroy()
-    this.pool = new this.Pool(assign(this.poolDefaults(config.pool || {}), config.pool))
-    this.pool.on('error', function(err) {
-      helpers.error('Pool2 - ' + err)
-    })
-    this.pool.on('warn', function(msg) {
-      helpers.warn('Pool2 - ' + msg)
+  initializePool: function(config, callback) {
+    var client = this
+    var promise = Promise.try(function() {
+      return Promise.try(function() {
+        if (client.pool) {
+          return client.destroy()
+        }
+      })
+      .then(function() {
+        client.pool = new client.Pool(assign(client.poolDefaults(config.pool || {}), config.pool))
+        client.pool.setMaxListeners(0) // Suppress warnings about too many event listeners
```

Comment:
I'm not sure. I can try it -- however I am not that familiar with Pool2's inner workings. I just went into the source for [pool#release](https://github.com/myndzi/pool2/blob/02e1ecd496857ba5799daa0c5d84e1b3ce694418/lib/pool.js#L184) and saw what events it emitted and went from there.

One of the issues I was trying to fix was that previously, the promise would never reject, but maybe that's because of how Pool2 is designed.

However, seeing issues like https://github.com/tgriesser/knex/issues/921 now makes me hesitant to change too much at once, so maybe it would be more prudent to revert the changes to this function, perhaps just keeping the ability to use a callback.

Might be good to get some input from @myndzi and @tgriesser on the "release resource" error handling story here.